templates/addons/aws/modules/vpc_flow_log/main.tf

```diff
@@ -0,0 +1,296 @@
+  bucket = aws_s3_bucket.flow_log_bucket.id
+
+  rule {
+    id     = "ExpireFlowLogs"
+    status = "Enabled"
+
+    filter {
+      prefix = "" # Apply to all objects
+    }
+
+    expiration {
+      days = var.log_retention_days
+    }
+  }
+}
+
+data "aws_iam_policy_document" "flow_logs_policy" {
+  statement {
+    sid       = "AWSLogDeliveryWrite"
+    actions   = ["s3:PutObject"]
+    resources = ["${aws_s3_bucket.flow_log_bucket.arn}/*"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["delivery.logs.amazonaws.com"]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [data.aws_caller_identity.current.account_id]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "s3:x-amz-acl"
+      values   = ["bucket-owner-full-control"]
+    }
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = ["arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:*"]
+    }
+  }
+
+  statement {
+    sid       = "AWSLogDeliveryAclCheck"
+    actions   = ["s3:GetBucketAcl", "s3:ListBucket"]
+    resources = [aws_s3_bucket.flow_log_bucket.arn]
+
+    principals {
+      type        = "Service"
+      identifiers = ["delivery.logs.amazonaws.com"]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [data.aws_caller_identity.current.account_id]
+    }
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = ["arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:*"]
+    }
+  }
+
+  statement {
+    principals {
+      type        = "*"
+      identifiers = ["*"]
+    }
+    effect = "Deny"
+    actions = [
+      "s3:*",
+    ]
+    resources = [
+      aws_s3_bucket.flow_log_bucket.arn,
+      "${aws_s3_bucket.flow_log_bucket.arn}/*",
+    ]
+    // Deny requests that are not using Secure Transport (HTTPS)
+    condition {
+      test     = "Bool"
+      variable = "aws:SecureTransport"
+      values   = ["false"]
+    }
+  }
+}
+
+resource "aws_s3_bucket_policy" "flow_logs_policy" {
+  bucket = aws_s3_bucket.flow_log_bucket.id
+  policy = data.aws_iam_policy_document.flow_logs_policy.json
+}
+
+# VPC Flow Log
+resource "aws_flow_log" "vpc_flow_log" {
+  log_destination      = aws_s3_bucket.flow_log_bucket.arn
+  log_destination_type = "s3"
+  traffic_type         = "ALL"
+  vpc_id               = var.vpc_id
+  log_format           = local.log_format
+  // Does NOT need IAM role here since s3 bucket policy handles the permissions
+
+  //Use Parquet format for better compression and cost efficiency when query with Athena
+  destination_options {
+    file_format                = "parquet"
+    hive_compatible_partitions = true
+    per_hour_partition         = true
+  }
+}
+
+// Athena table for VPC Flow Logs
```

Comment:
Could you also help add a small docs to wiki for security part and include this module?